compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -1465,7 +1465,7 @@ impl<'v> RootCollector<'_, 'v> {
 
                 // But even just declaring them must collect the items they refer to
                 // unless their generics require monomorphization.
-                if !self.tcx.generics_of(id.owner_id).requires_monomorphization(self.tcx)
+                if !self.tcx.generics_of(id.owner_id).own_requires_monomorphization()
```

Comment:
why change to `own_`? to improve perf?

Reply:
It's conceptually simpler but I don't feel strongly about it and can revert. Perf should be the ~same.